src/Components/Components/src/Routing/Router.cs

```diff
@@ -136,8 +136,10 @@ public async Task SetParametersAsync(ParameterView parameters)
             _onNavigateCalled = true;
             await RunOnNavigateAsync(NavigationManager.ToBaseRelativePath(_locationAbsolute), isNavigationIntercepted: false);
         }
-
-        Refresh(isNavigationIntercepted: false);
+        else
+        {
+            Refresh(isNavigationIntercepted: false);
+        }
```

Comment:
Could you write a test for it?

Reply:
I think our E2E Router tests could be enough. They will check whether `Router` renders the appropriate component. 
Or do you mean a test for checking that the component renders just once?

Reply:
We want to have a test that ensures we only render once. It's likely possible to author one as a unit test, rather than splurge for an E2E test.

Reply:
If we notice something in code like this, and we didn't catch it before, it means that there is a gap in our testing, so we want to 1) put a test in place to ensure the right behavior (before and after) and 2) prevent the behavior from regressing in the future

Reply:
Sorry I removed my previous comment. I just checked unit tests and agreed that we need a test for that and removed my question. and you were already typing the reply. 

Thanks!